src/lib/y2packager/repomd_downloader.rb

```diff
@@ -61,12 +64,20 @@ def product_repos
     # @return [Array<String>] List of paths pointing to the downloaded primary.xml.gz files,
     #   returns an empty list if the URL or the repository is not valid.
     #
-    def primary_xmls
-      return [] if product_repos.empty?
+    def primary_xmls(force = false)
+      # first check if there are any products defined in /media.1/products
+      repos = product_repos
+
+      if repos.empty?
+        return [] unless force
+
+        # if `force` is true then scan the repository at the root anyway
+        repos = { "" => "/" }
```

Comment:
well, I found quite strange to have empty string as key. Can we add some yast created name for repo? like product_root?

Reply:
it is just NP

Reply:
If you check the code below you will see that the name is not used in the loop at all. We could place anything here, only the path is relevant for scanning the products. We could use some generic name like "Repository" but as it is ignored anyway I decided to use an empty string as the most simple name. :smiley:

Reply:
Ok, I have added a comment to make it clear.